modMcf/src/org/aion/mcf/types/AbstractTransaction.java

```diff
@@ -54,8 +54,12 @@ public AbstractTransaction(byte[] nonce, Address receiveAddress, byte[] value, b
         this.to = receiveAddress;
         this.value = value;
         this.data = data;
-        // default type 0x01; reserve date for multi-type transaction
-        this.type = TransactionTypes.DEFAULT;
+        // setting the default type depending on use case
+        if (to == null) {
+            this.type = TransactionTypes.FVM_CREATE_CODE;
```

Comment:
How does the AVM contract create definition?
I thought the AVM contract create address destination is `null` too

Reply:
For AVM contracts the default value is incorrect. However, our unit tests have already been written to specify the deploy code and not use the default value when testing the AVM contract deployments. This change is mainly for the sake of the old FVM tests that were written without taking the transaction type into account.

Reply:
I would prefer fully removing the constructor that allows building transactions without specifying the type, but that would be a low priority refactoring that can be applied later.

Reply:
thanks for the explaination